### PR TITLE
NETOBSERV-1689: Add prom role and roleBindings to examples

### PR DESCRIPTION
**Pull request by Amoghrd**

### CODE REVIEW

File: examples/prometheus/role.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: netobserv-prom
+  namespace: netobserv
+rules:
+  - verbs:
+      - create
```
Comment (jpinsonneau) [inline examples/prometheus/role.yaml:8]: I'm wondering if we really need that `create` here.
Giving another try today

Comment (jpinsonneau) [reply examples/prometheus/role.yaml:8]: so I confirm we need the `create` as is but the behavior could be improved:
https://redhat-internal.slack.com/archives/C0VMT03S5/p1724752880713839?thread_ts=1715862728.898369&cid=C0VMT03S5

**Comment by jotak** [comment]
what's the use case for having these roles?

**Comment by jpinsonneau** [comment]
@jotak this is actually needed for developer perspective prometheus access.

The impersonation provided by OCP console somehow keep some of your admin role and bypass the Thanos check.
Creating a user and configuring OAuth to log in require this role.

Check the conversation with Simon https://redhat-internal.slack.com/archives/C0VMT03S5/p1724686028392639?thread_ts=1715862728.898369&cid=C0VMT03S5

**Comment by Amoghrd** [comment]
Should we merge this then? 
Or wait till @jotak creates a functionality in the operator to automatically deploy the role and the user only has to create the roleBinding

**Pull request closed without merging**